### PR TITLE
Adding first Go file & addition to readme

### PR DESCRIPTION
**Pull request by ThaWolfie**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package main
+
+import (
+	"fmt"
+	"os"
+)
+
+func main() {
+	{
```
Comment (bidesh) [inline main.go]: no need for the curly braces here. 
The braces is syntax that limits the scope of things declared inside to the point where the brace is closed. By scope, here, I mean the length of how long the variable will exist.
One of the reason you see it with every function declaration is  because you don't want to pollute the stuff outside of that function by things that you have done/declared inside. 
What the closing braces trigger is to "clean up" (in simpler terms) the variables that were declared inside. So, generally if you don't have syntax that needs it, you don't use it.

With that explanation, what would happen for the following? 

```
package main

import "fmt"

function main() {
    {
        x := 1
        fmt.Println(x)
    }
    fmt.Println(x)
}
```

**Comment by ThaWolfie** [comment]
Hi @bidesh I've amended the file can you review again?
